test/controllers/beneficial.owner.type.controller.spec.ts

```diff
@@ -293,8 +295,122 @@ describe("BENEFICIAL OWNER TYPE controller", () => {
+    //     .send({ [BeneficialOwnerTypeKey]: BeneficialOwnerTypeChoice.otherLegal });
+
+    //   expect(resp.status).toEqual(302);
+    //   expect(resp.header.location).toEqual(NEXT_PAGE_URL);
+    //   expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
+    //   expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.BENEFICIAL_OWNER_OTHER_WITH_PARAMS_URL);
+    // });
+
+    // test(`redirects to the ${config.BENEFICIAL_OWNER_GOV_PAGE} page`, async () => {
+    //   const resp = await request(app)
+    //     .post(config.BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL)
+    //     .send({ [BeneficialOwnerTypeKey]: BeneficialOwnerTypeChoice.government });
+
+    //   expect(resp.status).toEqual(302);
+    //   expect(resp.header.location).toEqual(NEXT_PAGE_URL);
+    //   expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
+    //   expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.BENEFICIAL_OWNER_GOV_WITH_PARAMS_URL);
+    // });
+
+    // test(`redirects to the ${config.MANAGING_OFFICER_PAGE} page`, async () => {
+    //   const resp = await request(app)
+    //     .post(config.BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL)
+    //     .send({ [BeneficialOwnerTypeKey]: ManagingOfficerTypeChoice.individual });
+
+    //   expect(resp.status).toEqual(302);
+    //   expect(resp.header.location).toEqual(NEXT_PAGE_URL);
+    //   expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
+    //   expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.MANAGING_OFFICER_WITH_PARAMS_URL);
+    // });
+
+    // test(`redirects to the ${config.MANAGING_OFFICER_CORPORATE_PAGE} page`, async () => {
+    //   const resp = await request(app)
+    //     .post(config.BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL)
+    //     .send({ [BeneficialOwnerTypeKey]: ManagingOfficerTypeChoice.corporate });
+
+    //   expect(resp.status).toEqual(302);
+    //   expect(resp.header.location).toEqual(NEXT_PAGE_URL);
+    //   expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);
+    //   expect(mockGetUrlWithParamsToPath.mock.calls[0][0]).toEqual(config.MANAGING_OFFICER_CORPORATE_WITH_PARAMS_URL);
+    // });
+
+    test(`renders the current page with error message when ${BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS} has been selected `, async () => {
+      mockIsActiveFeature.mockReturnValueOnce(true); // For FEATURE_FLAG_ENABLE_REDIS_REMOVAL
+
+      const resp = await request(app)
+        .post(config.BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL)
+        .send({ [BeneficialOwnerStatementKey]: BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS });
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(BENEFICIAL_OWNER_TYPE_PAGE_HEADING_ALL_IDENTIFIED_ALL_DETAILS);
+      expect(resp.text).toContain(ErrorMessages.SELECT_THE_TYPE_OF_BENEFICIAL_OWNER_YOU_WANT_TO_ADD);
+      expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(5);
```

Comment:
Rather than repeat this block of assertions four times, could we have a private function that extracts the logic and reduces the repetition and LOC?

Reply:
updated